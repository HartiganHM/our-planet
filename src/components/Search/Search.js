import React from 'react';
import './Search.css';

const Search = () => {
  return (
    <div className="Search">
      <input className="search-field" type="text" placeholder="Search" />
    </div>
  );
};

export default Search;
