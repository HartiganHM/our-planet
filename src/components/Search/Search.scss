@import '../../styles/mixins.scss';
@import '../../styles/variables.scss';

.Search {
  display: flex;
  @include margin(2);
  .search-field {
    flex: 1;
    @include padding(1);
    font-size: 1rem;
  }
}