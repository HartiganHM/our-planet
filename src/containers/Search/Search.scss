@import '../../styles/mixins.scss';
@import '../../styles/variables.scss';
.Search {
  display: flex;
  flex-direction: column;
  @include width(160);
  @include margin(2);
  @include padding(1);
  margin: auto;
  .search-field {
    flex: 1;
    @include padding(1);
    font-size: 1rem;
    &:focus {
      outline: none;
    }
  }
  .header-box {
    flex: 1;
    @include padding(1);
    color: $accent-color-dark;
    .header-title {
      @include contentDetailTitle();
    }
    .header-body {
      @include maxWidth(55);
      @include contentDetailBody();
    }
  }
}

@media screen and (max-width: 1599px) {
  .Search {
    @include width(128);
  }
}

@media screen and (max-width: 1279px) {
  .Search {
    @include width(96);
  }
}

@media screen and (max-width: 959px) {
  .Search {
    @include width(64);
  }
}

@media screen and (max-width: 639px) {
  .Search {
    @include width(32);
  }
}
