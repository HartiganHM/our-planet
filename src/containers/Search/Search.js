import React from 'react';
import { connect } from 'react-redux';
import FilterButtons from '../FilterButtons/FilterButtons';
import * as actions from '../../actions';
import { PropTypes } from 'prop-types';
import './Search.scss';

export const Search = props => {
  const { searchAnimals, animals, setSearchValue, searchValue } = props;
  const headerBox = (
    <div className="header-box">
      <div className="header-title">
        Search for animals of all shapes and sizes
      </div>
      <div className="header-body">
        Is your favorite animal endangered? Search below and click on an animal
        to learn more about them!
      </div>
    </div>
  );

  const handleChange = (inputValue, animals, property) => {
    searchAnimals(inputValue, animals, property);
    setSearchValue(inputValue);
  };

  return (
    <div className="Search" id="Search">
      {headerBox}
      <span className="wrapper">
        <input
          className="search-field"
          onChange={event => handleChange(event.target.value, animals, 'name')}
          type="text"
          placeholder="Search"
          value={searchValue}
        />
        <FilterButtons />
      </span>
    </div>
  );
};

export const mapStateToProps = store => {
  return {
    animals: store.animals,
    searchValue: store.searchValue,
  };
};

export const mapDispatchToProps = dispatch => {
  return {
    searchAnimals: (inputValue, animals, property) =>
      dispatch(actions.searchAnimals(inputValue, animals, property)),
    setSearchValue: inputValue => dispatch(actions.changeSearch(inputValue)),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Search);

Search.propTypes = {
  searchAnimals: PropTypes.func,
  animals: PropTypes.array,
  setSearchValue: PropTypes.func,
  searchValue: PropTypes.string,
};
