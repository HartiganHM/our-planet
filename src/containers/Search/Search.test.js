import React from 'react';
import { shallow } from 'enzyme';
import Search from './Search';

describe('Search tests', () => {
  it('Should exist', () => {
    const renderedSearch = shallow(<Search />);
    expect(renderedSearch).toBeDefined();
  });
});
