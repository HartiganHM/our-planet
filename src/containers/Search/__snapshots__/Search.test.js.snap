// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search tests Should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    animals={
        Array [
            Object {
              "display": true,
              "facts": "The Arctic fox is primarily a carnivore that lives inland, away from the coasts. They are dependent on the presence of smaller animals (most often lemmings) to survive.  Arctic foxes also hunt for sea birds, fish, and other marine life. Smaller rodent populations waver between times of abundance and scarcity, which leaves the Arctic fox vulnerable when these creatures are low in numbers.",
              "habitat": "Antarctica / Arctic",
              "height": "11 inches",
              "human_benefit": "The protection of the Arctic fox ensures the safety of a variety of other wildlife in the Arctic region.",
              "image": "arctic fox",
              "length": "18 to 27 inches",
              "name": "Arctic Fox",
              "population": "Several Hundred Thousand",
              "scientific_name": "Vulpes lagopus",
              "status": "Least Concern",
              "weight": "3 to 20 pounds",
            },
            Object {
              "display": true,
              "facts": "Often called the \\"polar wolf\\" or \\"white wolf,\\" Arctic wolves inhabit the Arctic regions of North America and Greenland. Thanks to its isolation, the Arctic wolf is not threatened by hunting and habitat destruction in the same way as its southern relatives.",
              "habitat": "Antarctica / Arctic",
              "height": "",
              "human_benefit": "Arctic wolves are carnivorous hunters. By nature they help to control the populations of other animals in the region like the musk ox, caribou and Arctic hares.",
              "image": "arctic wolf",
              "length": "3.2 to 5.9 feet",
              "name": "Arctic Wolf",
              "population": "",
              "scientific_name": "Vulpes lagopus",
              "status": "Least Concern",
              "weight": "70 to 175 pounds",
            },
          ]
    }
    searchAnimals={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <input
        className="search-field"
        onChange={[Function]}
        placeholder="Search"
        type="text"
/>,
      "className": "Search",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "search-field",
        "onChange": [Function],
        "placeholder": "Search",
        "type": "text",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <input
          className="search-field"
          onChange={[Function]}
          placeholder="Search"
          type="text"
/>,
        "className": "Search",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "search-field",
          "onChange": [Function],
          "placeholder": "Search",
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
